### ISSUE TITLE
Ensemble Model (BGRU + XGBoost)

### ISSUE THREAD

**Issue by pravanjang** [pravanjang/stock-prediction-system#16]
**Code to Generate:**
Complete script `models/ensemble.py` implementing ensemble:

**Required Class:**
```python
class BGRUXGBoostEnsemble:
    def __init__(self, bgru_model_path, n_features):
        """
        Ensemble Strategy:
        Model 1: Trained BGRU model (from Phase 2)
        Model 2: XGBoost classifier on same features
        Fusion: Weighted average or stacking
        """
        
    def train_xgboost(self, train_df, val_df):
        """
        Train XGBoost on technical + temporal + price action features
        Hyperparameters:
        - max_depth: 6
        - learning_rate: 0.1
        - n_estimators: 100
        - subsample: 0.8
        - colsample_bytree: 0.8
        """
        
    def predict_ensemble(self, test_df, method='weighted', weights=[0.6, 0.4]):
        """
        Ensemble methods:
        1. Weighted average: BGRU (60%) + XGBoost (40%)
        2. Stacking: Use logistic regression on predictions
        3. Voting: Majority vote
        """
        
    def optimize_weights(self, val_df):
        """Find optimal ensemble weights using validation set"""
        
    def save_ensemble(self, path='models/checkpoints/ensemble_model.pkl')
```

**Training Strategy:**
1. Load trained BGRU model
2. Train XGBoost on same features
3. Find optimal ensemble weights on validation set
4. Evaluate on test set

**Execution:**
```bash
python models/ensemble.py --bgru_model models/checkpoints/bgru_hybrid.pt --data data/processed/train_final.csv
```

**Output:**
- `models/checkpoints/xgboost_model.pkl`
- `models/checkpoints/ensemble_model.pkl`
- `models/ensemble_weights.json`
